Automattic/wp-calypso
# Track canvas mode changes in the Site Editor

Related to https://github.com/Automattic/dotcom-forge/issues/3301

## Proposed Changes

I added the sidebar click events on the Site Editor in https://github.com/Automattic/wp-calypso/pull/80953. This PR tracks events when the "canvas" is clicked, as well as navigation clicks. This is because users can start editing without clicking on the navigation, and we want to track that point. 

### Alternatives I did not go for

- Intercept the redux action; This cannot be done with [the current implementation](https://github.com/Automattic/wp-calypso/blob/6aae601ad42d6a43ba14b0b9cf57c8f0bea5ac41/apps/wpcom-block-editor/src/wpcom/features/tracking.js#L930) since [`setCanvasMode`](https://github.com/WordPress/gutenberg/blob/70ccdefe1fd0b02638f5a8ae4f1c03e6f7fe27cc/packages/edit-site/src/store/private-actions.js#L12) is private API and `unlock` function override JS object somehow.
- addEventListener to the DOM (`.edit-site-layout__canvas-container`); This might not be exactly the same as canvas mode changes.

## Testing Instructions



![e52dd3f431c383e3467fc7dd5b58ab46](https://github.com/Automattic/wp-calypso/assets/5287479/f253fab6-458f-483b-a7de-34f5e9fcbf44)

<img width="554" alt="Screen Shot 2023-09-04 at 10 43 40" src="https://github.com/Automattic/wp-calypso/assets/5287479/1cc1eff0-76d8-48fc-89d9-9b444fe787cf">

- Patch this PR to your sandbox with `cd apps/wpcom-block-editor/ && yarn dev --sync`
- Sandbox widgets.wp.com and your site
- Go to the Site Editor
- See if events fire with [Automattic/tracks-chrome-extension](https://github.com/Automattic/tracks-chrome-extension)

## Pre-merge Checklist
